code/modules/mob/living/simple_animal/bot/cleanbot.dm

```diff
@@ -285,29 +285,27 @@
 			mode = BOT_IDLE
 			return
 
-		if(loc == get_turf(target))
-			if(check_bot(target))
-				shuffle = TRUE //Shuffle the list the next time we scan so we dont both go the same way.
-				path = list()
-			else
-				UnarmedAttack(target, proximity_flag = TRUE) //Rather than check at every step of the way, let's check before we do an action, so we can rescan before the other bot.
-				if(QDELETED(target)) //We done here.
-					target = null
-					mode = BOT_IDLE
-					return
-
-		if(!length(path)) //No path, need a new one
-			//Try to produce a path to the target, and ignore airlocks to which it has access.
-			path = get_path_to(src, target, 30, id=access_card)
-			if(!bot_move(target))
-				add_to_ignore(target)
+		if(get_dist(src, target) <= 1)
+			UnarmedAttack(target, proximity_flag = TRUE) //Rather than check at every step of the way, let's check before we do an action, so we can rescan before the other bot.
+			if(QDELETED(target)) //We done here.
 				target = null
-				path = list()
+				mode = BOT_IDLE
 				return
+
+		if(target && path.len == 0 && (get_dist(src,target) > 1))
+			path = get_path_to(src, target, 30,id=access_card)
 			mode = BOT_MOVING
-		else if(!bot_move(target))
-			target = null
-			mode = BOT_IDLE
+			if(!path.len) //try to get closer if you can't reach the target directly
+				path = get_path_to(src, target, 30,1,id=access_card)
```

Comment:
Named parameter for 1 (and probably 30 as well)

Reply:
Why? Do you want me to change this for all bots or just cleanbots?

Reply:
https://github.com/tgstation/tgstation/blob/master/.github/guides/STYLE.md#prefer-named-arguments-when-the-meaning-is-not-obvious

Feel free to only change your own, but cleaning up the others would be good